app/controllers/products_controller.rb

```diff
@@ -16,6 +15,10 @@ def show
       redirect_to products_path
       return
     end   
+    
+    if session[:uid]
+      render :user_show
+    end
```

Comment:
I mean, if it works, it works, but I'm confused: where are we setting `session[uid]`? I know when someone logs in, it sets `session[:user_id]`, but wouldn't we also need to check that the product page the logged-in user is trying to view actually belongs to them and not another user?

Is that checking being done in some other method that then sets `session[:uid]`?